x-pack/platform/plugins/shared/embeddable_alerts_table/public/components/embeddable_alerts_table.test.tsx

```diff
@@ -0,0 +1,218 @@
+import { coreMock } from '@kbn/core/public/mocks';
+import type { AlertsTableProps } from '@kbn/response-ops-alerts-table/types';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { PERSISTED_TABLE_CONFIG_KEY_PREFIX } from '../constants';
+import { EmbeddableAlertsTable } from './embeddable_alerts_table';
+import { GENERAL_CASES_OWNER } from '@kbn/cases-plugin/common';
+import type { RuleTypeSolution } from '@kbn/alerting-types';
+
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+});
+
+const core = coreMock.createStart();
+core.http.get.mockResolvedValue([
+  {
+    id: 'o11y-rule-type',
+    solution: 'observability',
+  },
+]);
+const services = core as unknown as AlertsTableProps['services'];
+
+jest.mock('@kbn/response-ops-alerts-table', () => ({
+  AlertsTable: jest.fn(() => <div data-test-subj="alertsTable" />),
+}));
+const { AlertsTable: mockAlertsTable } = jest.requireMock('@kbn/response-ops-alerts-table');
+
+const TABLE_ID = `${PERSISTED_TABLE_CONFIG_KEY_PREFIX}-uuid`;
+
+describe('EmbeddableAlertsTable', () => {
+  afterEach(() => {
+    queryClient.clear();
+  });
+
+  it('should not render the alerts table until the rule types have been loaded', async () => {
+    render(
+      <QueryClientProvider client={queryClient}>
+        <EmbeddableAlertsTable
+          id={TABLE_ID}
+          timeRange={{
+            from: '2025-01-01T00:00:00.000Z',
+            to: '2025-01-01T01:00:00.000Z',
+          }}
+          services={services}
+        />
+      </QueryClientProvider>
+    );
+
+    expect(screen.getByRole('progressbar')).toBeInTheDocument();
+  });
+
+  it('should not render the alerts table if the rule types cannot been loaded', async () => {
+    core.http.get.mockImplementationOnce(() => {
+      throw new Error('Error loading rule types');
+    });
+    render(
+      <QueryClientProvider client={queryClient}>
+        <EmbeddableAlertsTable
+          id={TABLE_ID}
+          timeRange={{
+            from: '2025-01-01T00:00:00.000Z',
+            to: '2025-01-01T01:00:00.000Z',
+          }}
+          services={services}
+        />
+      </QueryClientProvider>
+    );
+
+    expect(await screen.findByText('Cannot load rule types')).toBeInTheDocument();
+  });
+
+  it('should show a missing auth prompt if the solution rule types are not accessible', async () => {
+    render(
+      <QueryClientProvider client={queryClient}>
+        <EmbeddableAlertsTable
+          id={TABLE_ID}
+          solution="security"
+          timeRange={{
+            from: '2025-01-01T00:00:00.000Z',
+            to: '2025-01-01T01:00:00.000Z',
+          }}
+          services={services}
+        />
+      </QueryClientProvider>
+    );
+
+    expect(await screen.findByText('Missing alerting authorizations')).toBeInTheDocument();
+  });
+
+  it('should render the alerts table with the correct time range and base props', async () => {
```

Comment:
Let's add a test where the time range and `filtersQuery` are combined correctly.